docs/architecture/modern-web-apps-azure/develop-asp-net-core-mvc-apps.md

```diff
@@ -381,6 +400,8 @@ public async Task<IActionResult> Put(int id, [FromBody]Author author)
 
 You can read more about implementing filters and download a working sample from the MSDN Magazine article, [Real-World ASP.NET Core MVC Filters](/archive/msdn-magazine/2016/august/asp-net-core-real-world-asp-net-core-mvc-filters).
 
+If you find that you have a number of common responses from APIs based on common scenarios like validation errors (Bad Request), resource not found, and server errors, you might consider using a *result* abstraction. The result abstraction would be returned by services consumed by API endpoints, and the controller action or endpoint would use a filter to translate these into IActionResults. One example of the result pattern can be found in this [Result NuGet package](https://www.nuget.org/packages/Ardalis.Result/) and [its associated MVC filter](https://www.nuget.org/packages/Ardalis.Result.AspNetCore/).
```

Comment:
```suggestion
If you find that you have a number of common responses from APIs based on common scenarios like validation errors (Bad Request), resource not found, and server errors, you might consider using a *result* abstraction. The result abstraction would be returned by services consumed by API endpoints, and the controller action or endpoint would use a filter to translate these into IActionResults.
```

Reply:
@IEvangelist  I am not sure about this pattern. What do you think about this para?

Reply:
It's a well-known pattern. It's functionally equivalent to other languages' Result or Maybe types:

https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/results

https://wiki.haskell.org/Maybe

I can reference the pattern in the book and move specific implementation details to GitHub if you like.

Reply:
Perfect, thanks for the details. It will be great if you can explain this pattern a little more and add implementation details in GitHub's Wiki.